src/bin/poi2mimir.rs

```diff
@@ -0,0 +1,220 @@
+        weight: weight,
+        zip_codes: vec![],
+        poi_type: poi_type,
+        properties: poi
+            .properties
+            .into_iter()
+            .map(|p| Property::from(p))
+            .collect(),
+        address: addr,
+        country_codes: country_codes,
+        names: I18nProperties::default(),
+        labels: I18nProperties::default(),
+        distance: None,
+    };
+
+    Ok(poi)
+}
+
+fn import_pois(
+    rubber: &mut Rubber,
+    index: TypedIndex<Poi>,
+    dataset: &str,
+    file: &PathBuf,
+) -> Result<(), mimirsbrunn::Error>
+where
+{
+    info!("Add data in elasticsearch db.");
+
+    let model = NavitiaModel::try_from_path(file)?;
+    let poi_types = model.poi_types;
+
+    // Note: We're ignoring those POIs that fail to be enriched.
+    let pois: Vec<_> = model
+        .pois
+        .into_iter()
+        .filter_map(|(id, poi)| {
+            into_mimir_poi(poi, &poi_types, rubber)
+                .map_err(|err| warn!("Could not extract information for POI '{}': {}", id, err))
+                .ok()
+        })
+        .collect(); // TODO Can we get rid of collect, and chain with the following rubber...?
+
+    let count = rubber
+        .bulk_index(&index, pois.into_iter())
+        .map_err(|err| format_err!("Failed bulk insertion {}", err))?;
+
+    info!("importing POIs: {} POIs added.", count);
+
+    rubber
+        .publish_index(dataset, index)
+        .map_err(|err| format_err!("Failed to publish index {}.", err))?;
+
+    Ok(())
+}
+
+/// This function initializes the ES context: It creates an index for this dataset,
+/// and then import the POIs in it.
+fn index_poi(
+    cnx_string: &str,
+    dataset: &str,
+    file: &PathBuf,
+    nb_shards: usize,
+    nb_replicas: usize,
+) -> Result<(), mimirsbrunn::Error>
+where
+{
+    let mut rubber = Rubber::new(cnx_string);
+    rubber.initialize_templates()?;
+
+    let settings = IndexSettings {
+        nb_shards: nb_shards,
+        nb_replicas: nb_replicas,
+    };
+
+    let index = rubber.make_index(dataset, &settings)?;
+
+    import_pois(&mut rubber, index, dataset, file)
+}
+
+#[derive(StructOpt, Debug)]
+struct Args {
+    /// POI file
+    #[structopt(short = "i", long = "input", parse(from_os_str))]
+    input: PathBuf,
+
+    /// Elasticsearch parameters.
+    #[structopt(
+        short = "c",
+        long = "connection-string",
+        default_value = "http://localhost:9200/munin"
+    )]
+    connection_string: String,
+
+    /// Name of the dataset.
+    /// A dataset is a label, that can be used for filtering the data.
+    #[structopt(short = "d", long = "dataset", default_value = "fr")]
+    dataset: String,
+
+    /// Number of threads to use
+    #[structopt(
+        short = "t",
+        long = "nb-threads",
+        raw(default_value = "&DEFAULT_NB_THREADS")
+    )]
+    nb_threads: usize,
+
+    /// Number of shards for the es index
+    #[structopt(short = "s", long = "nb-shards", default_value = "5")]
```

Comment:
```suggestion
    #[structopt(short = "s", long = "nb-shards", default_value = "1")]
```
for less than 1 millions doc 1 shard is enough, most of the time we won't have 1 million POI per dataset